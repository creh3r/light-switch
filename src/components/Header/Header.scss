.header {
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: $gutter-times-four;
  width: 100%;
  padding: $gutter 0;
  background-color: map-get($theme-colors, greyLightened);

  &__title {
    font-size: $font-size-header;
    text-transform: uppercase;
  }

  &__children {
    @media (min-width: 768px) {  
      position: absolute;
      right: $gutter;
    }
  }
}