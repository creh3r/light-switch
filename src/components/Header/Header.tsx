import React from 'react';

type HeaderProps = {
  title: string,
  children: React.ReactNode,
};

/**
 * Render the game header
 *
 * @param title - The header title
 * @param children - The children
 */
function Header({ title, children }: HeaderProps) {
  return (
    <div className="header">
      {title}
      <div className="header__children">
        { children }
      </div>
    </div>
  );
}

export default Header;
