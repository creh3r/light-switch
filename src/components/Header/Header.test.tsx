// eslint-disable-next-line import/no-unresolved
import { shallow } from 'enzyme';
import React from 'react';
import Header from './Header';

describe('<Header /> component', () => {
  it('... should match snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
