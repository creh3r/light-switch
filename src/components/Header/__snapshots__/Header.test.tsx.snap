// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component ... should match snapshot 1`] = `
<div
  className="header"
>
  <div
    className="header__children"
  />
</div>
`;
